src/main/java/net/minecraftforge/client/ClientCommandHandler.java

```diff
@@ -0,0 +1,312 @@
+            resultBuilder = LiteralArgumentBuilder.literal(sourceLiteral.getLiteral());
+        }
+        else if (sourceNode instanceof RootCommandNode<?>)
+        {
+            CommandNode<T> resultNode = new RootCommandNode<>();
+            mergeCommandNode(sourceNode, resultNode, sourceToResult, canUse, execute, sourceToResultSuggestion);
+            return resultNode;
+        }
+        else
+        {
+            throw new IllegalStateException("Node type " + sourceNode + " is not a standard node type");
+        }
+
+        if (sourceNode.getCommand() != null)
+        {
+            resultBuilder.executes(execute);
+        }
+
+        if (sourceNode.getRedirect() != null)
+        {
+            resultBuilder.redirect(toResult(sourceNode.getRedirect(), sourceToResult, canUse, execute, sourceToResultSuggestion));
+        }
+        
+        CommandNode<T> resultNode = resultBuilder.build();
+        mergeCommandNode(sourceNode, resultNode, sourceToResult, canUse, execute, sourceToResultSuggestion);
+        return resultNode;
+    }
+
+    /**
+     * Always try to execute the cached parsing of client message as a command. Requires that the execute field of the commands to be set to send to server so that they aren't
+     * treated as client command's that do nothing.
+     * 
+     * {@link net.minecraft.commands.Commands#performCommand(CommandSourceStack, String)} for reference
+     * 
+     * @param sendMessage
+     *            the chat message
+     * @return false leaves the message to be sent to the server, true means it should be caught before {@link LocalPlayer#chat(String)}
+     */
+    public static boolean sendMessage(String sendMessage)
+    {
+        ClientCommandSourceStack source = getSource();
+
+        StringReader reader = new StringReader(sendMessage);
+
+        if (!reader.canRead() || reader.read() != '/')
+        {
+            return false;
+        }
+
+        try
+        {
+            commands.execute(reader, source);
+        }
+        catch (CommandRuntimeException execution)// Probably thrown by the command
+        {
+            Minecraft.getInstance().player.sendMessage(new TextComponent("").append(execution.getComponent()).withStyle(ChatFormatting.RED), Util.NIL_UUID);
+        }
+        catch (CommandSyntaxException syntax)// Usually thrown by the CommandDispatcher
+        {
+            Minecraft.getInstance().player.sendMessage(
+                    new TextComponent("").append(ComponentUtils.fromMessage(syntax.getRawMessage())).withStyle(ChatFormatting.RED), Util.NIL_UUID);
+            if (syntax.getInput() != null && syntax.getCursor() >= 0)
+            {
+                int position = Math.min(syntax.getInput().length(), syntax.getCursor());
+                MutableComponent details = new TextComponent("").withStyle(ChatFormatting.GRAY);
+                details.getStyle().withClickEvent(new ClickEvent(ClickEvent.Action.SUGGEST_COMMAND, reader.getString()));
```

Comment:
This won't work, as `Style#withClickEvent` returns a new `Style` (as they are immutable objects).